src/storage/src/storage_state.rs

```diff
@@ -550,13 +550,24 @@ impl<'w, A: Allocate> Worker<'w, A> {
                     resumption_frontier,
                 });
             }
+            AsyncStorageWorkerResponse::UpdatedSinkDesc(id, sink_desc) => {
+                // NOTE: If we want to share the load of async processing we
+                // have to change `handle_storage_command` and change this
+                // assert.
+                assert!(
+                    self.timely_worker.index() == 0,
+                    "only worker #0 is doing async processing"
+                );
+                internal_cmd_tx
+                    .broadcast(InternalStorageCommand::CreateSinkDataflow(id, sink_desc));
+            }
         }
     }
 
     /// Entry point for applying an internal storage command.
     pub fn handle_internal_storage_command(
         &mut self,
-        internal_cmd_tx: &mut dyn InternalCommandSender,
```

Comment:
Maybe we should declare that all future commands go via the async worker? Seems less susceptible to races if all commands take the same path and get processed in order...

Reply:
I like that! Though I'd like to get other peoples opinion on that as well. @petrosagg?